### DISCUSSION TITLE
"Secondary Weapon" turrets - Exact syntax please?

### DISCUSSION THREAD

**Discussion by esksp** [endless-sky/endless-sky#10716] [Q&A] [answered]
I created **a turret which is a "Secondary Weapon"**, so it won't fire when you fire your usual guns (because it would drain your energy pretty quickly and needlessly).
There are already at least two "Secondary Weapon" turrets I could find in the game, and I took the "Firestorm Battery" as an example.
So far so good.

Unfortunately once in game, **it works, but not as a "Secondary Weapon", it just acts like a normal turret**.
It does not show the little "ammo" icon on the lower right, and does not react to any of the "Secondary Weapon" hotkeys.
It does fire when you fire your main guns though, like a normal turret.

What did I do wrong?

```
outfit "Planetary Destruction Gun"
	category "Secondary Weapons"
	cost 62000000
	thumbnail "outfit/firestorm battery"	# --- Temp ---
	"mass" 80
	"outfit space" -80
	"weapon capacity" -40
	"turret mounts" -1
	"spinal mount" -1
	"required crew" 0
	"unplunderable" 1
	weapon
		sprite "projectile/ionic turret"	# --- Temp ---
			"frame rate" 20
		"fade out" 10
		...
		...

```

**Comment by TheGiraffe3** [comment, marked as answer]
Did you give it the `icon` attribute?

**Reply by esksp** [reply]
I missed that! Thanks!

But is that why it doesn't act as a "secondary weapon"?
I would had expected that it just doesn't show a munition icon, but otherwise works as expected.

**Reply by danaris** [reply]
In order for it to act as a secondary weapon, you need to be able to select it from your list of secondary weapons; the only way to tell which it is in that list is the icon.

**Reply by esksp** [reply]
I see. I would expect for it to just show an empty space since I actually select it by hotkey.
But I tried it and adding an icon fixed everything, it now acts like a secondary weapon.
Thanks guys!

**Comment by esksp** [comment]
Hmm. There are also **"secondary weapons" which don't have an "icon" attribute**. I'm thinking of the "Ember Tear" here.
Well, it's a gun, so I guess it is supposed to simply fire with all your other guns, but what's the point of making it a "secondary weapon" in this case? Inquiring minds want to know...

Note I haven't met it in-game yet, so I might be missing something obvious. Also please don't spoil me too much... :-)

**Reply by TheGiraffe3** [reply]
I'd guess so that you know it uses fuel and a lot of energy.

**Reply by Hecter94** [reply]
Ember Tear isn't player-obtainable, so there's no reason for it to have an icon.
